ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt

```diff
@@ -115,19 +118,26 @@ constructor(
     activeSurvey = null
   }
 
-  suspend fun getSurveySummaries(user: User): Flow<List<Survey>> =
-    try {
-      val surveys =
-        withTimeout(LOAD_REMOTE_SURVEY_SUMMARIES_TIMEOUT_MILLIS) {
-          Timber.d("Loading survey list from remote")
-          remoteDataStore.loadSurveySummaries(user)
-        }
-      listOf(surveys).asFlow()
-    } catch (e: Throwable) {
-      Timber.d(e, "Failed to load survey list from remote")
-      offlineSurveys
+  fun getSurveyList(user: User): Flow<List<Pair<Survey, Boolean>>> =
+    @OptIn(ExperimentalCoroutinesApi::class)
+    networkManager.networkStatusFlow.flatMapLatest { networkStatus ->
```

Comment:
@shobhitagarwal1612 Here's the basic idea. I tested this and it appears to be working well. I'll fix TODOs and tests and send an updated PR soon.

Reply:
Nice!